controller/src/beerocks/master/db/node.cpp

```diff
@@ -297,3 +297,89 @@ bool node::set_type(beerocks::eType type_)
     }
     return false;
 }
+
+bool node::link_metrics_data::add_transmitter_link_metric(
+    std::shared_ptr<ieee1905_1::tlvTransmitterLinkMetric> TxLinkMetricData)
+{
+    //  interface_pair_info_length() returns the length in bytes (number of elements * sizeof(sInterfacePairInfo).
+    size_t info_size = TxLinkMetricData->interface_pair_info_length() /
+                       sizeof(ieee1905_1::tlvTransmitterLinkMetric::sInterfacePairInfo);
+
+    for (size_t i = 0; i < info_size; i++) {
+        auto info_tuple = TxLinkMetricData->interface_pair_info(i);
+
+        if (!std::get<0>(info_tuple)) {
+            LOG(ERROR) << "add_transmitter_link_metric getting operating class entry has failed!";
+            return false;
+        }
+        auto &InterfacePairInfo = std::get<1>(info_tuple);
+        transmitterLinkMetrics.push_back(InterfacePairInfo);
+
+        LOG(DEBUG) << "adding tlvTransmitterLinkMetric data to list"
+                   << " phy_rate = " << int(InterfacePairInfo.link_metric_info.phy_rate);
+    }
+    return true;
+}
+
+bool node::link_metrics_data::add_receiver_link_metric(
+    std::shared_ptr<ieee1905_1::tlvReceiverLinkMetric> RxLinkMetricData)
+{
+    //  interface_pair_info_length() returns the length in bytes (number of elements * sizeof(sInterfacePairInfo).
+    size_t info_size = RxLinkMetricData->interface_pair_info_length() /
+                       sizeof(ieee1905_1::tlvReceiverLinkMetric::sInterfacePairInfo);
+
+    for (size_t i = 0; i < info_size; i++) {
+        auto info_tuple = RxLinkMetricData->interface_pair_info(i);
+
+        if (!std::get<0>(info_tuple)) {
+            LOG(ERROR) << "add_receiver_link_metric getting operating class entry has failed!";
+            return false;
+        }
+        auto &InterfacePairInfo = std::get<1>(info_tuple);
+        receiverLinkMetrics.push_back(InterfacePairInfo);
+
+        LOG(DEBUG) << "adding tlvReceiverLinkMetric data to list"
+                   << " rssi_db = " << int(InterfacePairInfo.link_metric_info.rssi_db);
+    }
+    return true;
+}
+
+bool node::ap_metrics_data::add_ap_metric_data(std::shared_ptr<wfa_map::tlvApMetric> ApMetricData)
+{
+    bssid                               = ApMetricData->bssid();
+    channel_utilization                 = ApMetricData->channel_utilization();
+    number_of_stas_currently_associated = ApMetricData->number_of_stas_currently_associated();
+
+    if (ApMetricData->estimated_service_parameters().include_ac_be) {
+        for (int i = 0; i < 3; i++) {
+            std::copy(ApMetricData->estimated_service_info_field_ac_be(),
+                      ApMetricData->estimated_service_info_field_ac_be() + 3,
+                      estimated_service_info_field_ac_be);
+        }
```

Comment:
This loop makes exactly the same copy 3 times! In other words, the loop is not needed.